internal/validator/main.go

```diff
@@ -0,0 +1,144 @@
+
+func init() {
+	tokenFile = flag.String("token-file", "token.json", "path to token file")
+}
+
+type TokenFile map[string]*token.Token
+
+func main() {
+
+	flag.Parse()
+	// read token file
+	tokens, err := os.ReadFile(*tokenFile)
+	if err != nil {
+		log.Errorf("cannot read token file: %s", err)
+		os.Exit(1)
+	}
+	isValid := json.Valid(tokens)
+	if !isValid {
+		log.Errorf("invalid json: %s", tokens)
+		os.Exit(1)
+	}
+
+	var metadata = TokenFile{}
+
+	decoder := json.NewDecoder(strings.NewReader(string(tokens)))
+	decoder.DisallowUnknownFields()
+	err = decoder.Decode(&TokenFile{})
+	if err != nil {
+		log.Errorf("cannot decode token file: %s", err)
+		os.Exit(1)
+	}
+
+	if err := json.Unmarshal(tokens, &metadata); err != nil {
+		fmt.Println("cannot unmarshal token file:", err)
+		os.Exit(1)
+	}
+
+	log.Infof("token file is semantically valid")
+
+	networkNode := "mainnet,lb"
+
+	networkNodeSplit := strings.Split(networkNode, ",")
+	networkStr, node := networkNodeSplit[0], networkNodeSplit[1]
+	network := common.LoadNetwork(networkStr, node)
+
+	exchangeClient, err := exchange.NewExchangeClient(network)
+	if err != nil {
+		log.Errorf("cannot create mainnet exchange client: %s", err)
+		os.Exit(1)
+	}
+
+	networkStr = "testnet"
+	network = common.LoadNetwork(networkStr, node)
+	testnetExchangeClient, err := exchange.NewExchangeClient(network)
+	if err != nil {
+		log.Errorf("cannot create testnet exchange client: %s", err)
+		os.Exit(1)
+	}
+
+	finder := MarketFinder{
+		exchangeClient: map[string]exchange.ExchangeClient{
+			"mainnet": exchangeClient,
+			"testnet": testnetExchangeClient,
+		},
+	}
+	for key, m := range metadata {
+		if strings.Contains(strings.ToLower(key), "test") {
+			continue
+		}
+		if m.Denom != "" {
+			for _, market := range m.InjectiveMarkets {
+				if _, ok := TestsMarkets[market]; ok {
+					continue
+				}
+				_, err := finder.findMarketOnAllNetwork(context.TODO(), market)
+				if err != nil {
+					log.Fatalf("%s: market %s not found: %s", key, market, err)
+				} else {
+					log.Infof("%s: market %s found", key, market)
+				}
+			}
+		}
+	}
+
+	log.Infof("token file content is valid")
+}
+
+type MarketFinder struct {
+	exchangeClient map[string]exchange.ExchangeClient
+}
+
+func (f MarketFinder) findMarketOnAllNetwork(ctx context.Context, market string) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 1000*time.Second)
+	defer cancel()
+
+	for key, client := range f.exchangeClient {
+		spotMarketInfo := injective_spot_exchange_rpcpb.MarketResponse{}
+		derivativeMarketInfo := injective_derivative_exchange_rpcpb.MarketResponse{}
+		spotMarketInfo, err := client.GetSpotMarket(ctx, market)
+		if err != nil {
+			derivativeMarketInfo, err = client.GetDerivativeMarket(ctx, market)
+			if err != nil &&
+				!strings.Contains(err.Error(), "not found") &&
+				!strings.Contains(err.Error(), "502") {
+				return "", fmt.Errorf("cannot get market %s on %s: %s", market, key, err)
+			}
```

Comment:
This is just a heads up, I am not expecting this to be solved as it would take some time but we should handle the error here properly with `errors.Is`. 

Ideally, since we own this repository (`InjectiveLabs/sdk-go`), we should create setinel errors and make them public so whoever imports this package can handle errors accordingly as below:
```
derivativeMarketInfo, err = client.GetDerivativeMarket(ctx, market)
if err != nil &&
	!errors.Is(err, exchange.ErrNotFound) &&
	!errors.Is(err, , exchange.ErrBadGateway) {
	return "", fmt.Errorf("cannot get market %s on %s: %s", market, key, err)
}
```